tests/e2e/push_test.go

```diff
@@ -206,17 +182,70 @@ func TestPushForceAndIgnoreIfExists(t *testing.T) {
 	if err := cmd.Run(); err == nil {
 		t.Errorf("Expected error")
 	}
-
-	if stdout.String() != "" {
-		t.Errorf("Expected stdout to be empty, but got %q", stdout.String())
-	}
+	assertEmptyOutput(t, stdout, nil)
 
 	expectedErrorMessage := "The --force and --ignore-if-exists flags are mutually exclusive and cannot be specified together."
 	if !strings.HasPrefix(stderr.String(), expectedErrorMessage) {
 		t.Errorf("Expected stderr to begin with %q, but got %q", expectedErrorMessage, stderr.String())
 	}
 }
 
+func TestPushRelative(t *testing.T) {
+	t.Log("Test push action with --relative flag")
+
+	name := "test-push-relative"
+	dir := "charts"
+	chartName := "foo"
+	chartVer := "1.2.3"
+	filename := fmt.Sprintf("%s-%s.tgz", chartName, chartVer)
+
+	setupRepo(t, name, dir)
+	defer teardownRepo(t, name)
+
+	// set a cleanup in beforehand
+	defer removeObject(t, name, dir+"/"+filename)
+
+	cmd, stdout, stderr := command(fmt.Sprintf("helm s3 push --relative testdata/%s %s", filename, name))
+	if err := cmd.Run(); err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	assertEmptyOutput(t, stdout, stderr)
+
+	tmpdir, err := ioutil.TempDir("", t.Name())
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	defer os.RemoveAll(tmpdir)
+
+	indexFile := filepath.Join(tmpdir, "index.yaml")
+
+	if err := mc.FGetObject(name, dir+"/index.yaml", indexFile, minio.GetObjectOptions{}); err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	idx, err := repo.LoadIndexFile(indexFile)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	v, err := idx.Get(chartName, chartVer)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	expected := []string{filename}
+	if diff := cmp.Diff(expected, v.URLs); diff != "" {
+		t.Errorf("mismatch (-want +got):\n%s", diff)
+	}
+
+	os.Chdir(tmpdir)
+	cmd, stdout, stderr = command(fmt.Sprintf("helm fetch %s/%s --version %s", name, chartName, chartVer))
```

Comment:
`os.Chdir` can cause problems.

Could you please either add `--destination` flag to helm command or use something like

```
	wd, _ := os.Getwd()
	os.Chdir(tmpdir)
	defer os.Chdir(wd)
```